pkg/apis/bundle/v1alpha1/builder_types.go

```diff
@@ -71,6 +71,20 @@ type ComponentBuilder struct {
 	// it should be representable as either YAML or JSON.
 	ObjectFiles []File `json:"objectFiles,omitempty"`
 
+	// Templates that are specified via a File-URL. The process of inlining a
+	// component turns template files into objects with template.  During the
+	// inline process, if the file is YAML-formatted and contains multiple
+	// objects in the YAML-doc, the objects will be split into separate inline
+	// objects. In other words, one object file may result in multiple objects.
+	//
+	// Each object file must be parsable into a Struct: In other words,
+	// it should be representable as either YAML or JSON.
+	TemplateFiles []File `json:"templateFiles,omitempty"`
```

Comment:
I was imagining something like

```
TemplateFiles: 
- templateType: foo
   url: bar.yaml
```

Although that doesn't quite work with File type. So maybe instead

```
TemplateFiles: 
- templateType: foo
   files: 
   - url: bar.yaml
```

with 

```
type TemplateFile struct {
 TemplateType TemplateType
  Files []File
}
```

This would allow the possibility of specifying multiple different template types in on Component, which I rather like for flexibility